src/checks/requiredTypeScript.ts

```diff
@@ -0,0 +1,224 @@
+import * as core from '@actions/core'
+import * as CommentedJSON from 'comment-json'
+
+import { WebhookEventMap } from '@octokit/webhooks-types'
+
+import { github } from '../infrastructure/github'
+import * as fs from '../infrastructure/fs'
+import { isBot } from '../conditions/triggeredByBot'
+
+import Check from './check'
+import ignore, { Ignore as IgnoredFileFilter } from 'ignore'
+
+type TsConfig = {
+  compilerOptions?: {
+    allowUnreachableCode?: boolean
+    noImplicitAny?: boolean
+  }
+}
+
+const requiredTypeScript: Check = {
+  name: 'required-typescript',
+  optional: true,
+  async run(): Promise<boolean> {
+    if (github.context.eventName === 'pull_request') {
+      return checkPullRequest()
+    }
+
+    core.info(
+      'Required TypeScript only runs on "pull_request" events. Skipping...'
+    )
+    return true
+  },
+}
+export default requiredTypeScript
+
+async function checkPullRequest(): Promise<boolean> {
+  const pullPayload = github.context.payload as WebhookEventMap['pull_request']
+  const pr = await github.getPullRequest(pullPayload.pull_request.number)
+
+  const prUser = pr.user?.login || ''
+
+  if (isBot(prUser)) {
+    core.info(`PR is from bot user ${prUser}. Skipping check`)
+    return true
+  }
+
```

Comment:
Wouldn't it be a good idea to do and early check verifying if the codebase (or changes) contain JS files, that way we can add an explicit exit saying there is nothing to do because no JS involved.
Otherwise it has to go through the next two checks before outputting a ` 'OK! No forbidden JS additions/changes or missing "tsconfig.json" settings'`.